### ISSUE TITLE
Metric to minimize in fmin

### ISSUE THREAD

**Issue by guillaume-chevalier** [hyperopt/hyperopt#308]
Hi, 

Let's say in a neural network that someone adds L2 weight decay regularization. This increases the final loss, but despite all, it may very well improve results obtained. I do not want to return the "negative of the accuracy" in the "loss" key of the returned dict of my function to minimize because that would be unclean and misleading. 

Could you (or a contributor) please add a positional argument to `fmin` to be able to pass the name of the key that we want to minimize? Still, one problem persists: I would need to return the negative of the accuracy. Ideally, it would be good to also have a function of the other one to maximize the provided key rather than to minimize it. 

Thanks a lot!

**Comment by michaelmior** [comment]
I'm not sure I understand what's unclean and misleading about just returning the negative of the accuracy as loss. With an appropriate comment to explain what's going on, this seems quite reasonable to me and changing the interface to fmin feels unnecessary.

**Comment by guillaume-chevalier** [comment]
I would still suggest is to have an "fmax" function and the possibility to change the key for having some "clean code" to minimise the number of "wtf" someone would have if he read the code. Anyway, feel free to close this issue if this isn't easy to implement according to how "fmin" is already coded. At least my suggested feature would not break the current API and would be backward-compatible with older versions because of the added positional argument.

**Comment by michaelmior** [comment]
Closing since there is a solution to this. Feel free to submit a PR if you still want this feature.
